### PR TITLE
Defect/de3351 deleting user first name

### PR DESCRIPTION
**Pull request by silbermm**

### CODE REVIEW

File: Gateway/MinistryPlatform.Translation.Test/Services/ContactServiceTest.cs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Reflection;
```
Comment (andycanterbury) [inline Gateway/MinistryPlatform.Translation.Test/Services/ContactServiceTest.cs]: Is this using... used?

File: Gateway/MinistryPlatform.Translation.Test/Services/ContactServiceTest.cs [hunk 2]
```diff
@@ -485,6 +486,83 @@ public void ShouldGetOtherHouseholdMembers()
...
+            const int fakeContactId = 546879;
+
+            try
+            {
+                _fixture.UpdateContact(fakeContactId, fakeProfileDict);
+                Assert.Fail("Application Exception wasn\'t thrown");
+            }
+            catch (ApplicationException e)
+            {
+                Assert.True(e.Message.Contains("First_Name was not found or was null"));
```
Comment (andycanterbury) [inline Gateway/MinistryPlatform.Translation.Test/Services/ContactServiceTest.cs]: I guess this works, but you can also Assert.Throws()

File: Gateway/crds-angular/Controllers/API/GroupLeaderController.cs [hunk 3]
```diff
@@ -55,7 +55,7 @@ public GroupLeaderController(IGroupLeaderService groupLeaderService, IUserImpers
                     {                        
 
                         _groupLeaderService.SaveReferences(profile).Zip<int, IList<Unit>, int>(_groupLeaderService.SaveProfile(token, profile),
-                                                     (int first, IList<Unit> second) => first).ToTask();
+                                                     (int first, IList<Unit> second) => first).ToTask().Wait();
```
Comment (andycanterbury) [inline Gateway/crds-angular/Controllers/API/GroupLeaderController.cs]: what are first and second? Also, is there any benefit in making this a task if you are always going to wait for it to finish before continuing?

File: Gateway/MinistryPlatform.Translation.Test/Services/ContactServiceTest.cs (hunk 1 above)
Comment (silbermm) [inline Gateway/MinistryPlatform.Translation.Test/Services/ContactServiceTest.cs]: fixed

File: Gateway/crds-angular/Controllers/API/GroupLeaderController.cs (hunk 3 above)
Comment (silbermm) [inline Gateway/crds-angular/Controllers/API/GroupLeaderController.cs]: first is the result of the first observable, second is the result of the second obserable

File: Gateway/crds-angular.test/Services/GroupLeaderServiceTest.cs [hunk 4]
```diff
@@ -127,7 +151,14 @@ public void ShouldRethrowExceptionWhenUserServiceThrows()
             const string fakeToken = "letmein";
             const int fakeUserId = 98124;
             var leaderDto = GroupLeaderMock();
-            _personService.Setup(m => m.SetProfile(fakeToken, It.IsAny<Person>()));
+            var fakePerson = new Person
```
Comment (andycanterbury) [inline Gateway/crds-angular.test/Services/GroupLeaderServiceTest.cs]: This same code is here 4 times. Maybe pull it into a method to DRY it up?

**Pull request merged**